### ISSUE TITLE
Lose inference shape in onnx exported graph when using Resize augmentation

### ISSUE THREAD

**Issue by duc12111** [kornia/kornia#1969]
### Describe the bug

Kornia augmentation Resize loses the inference shape of onnx exported graph. I wonder if this can be fixed.

### Reproduction steps

```bash
>>> import kornia as K
>>> import cv2
>>> import torch
>>> import torch.nn as nn
>>> import onnx
>>> img = cv2.imread("test_image.jpg")
>>> img = K.image_to_tensor(img)
>>> img = img.float()/255.
>>> model = nn.Sequential(K.augmentation.Resize(size = [72,170]))
>>> torch.onnx.export(model, img, "model.onnx", verbose=True, export_params=True)
```

### Expected behavior

/home/duc/anaconda3/lib/python3.9/site-packages/kornia/geometry/transform/imgwarp.py:334: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  KORNIA_CHECK(points_src.shape == points_dst.shape, "Source data shape must match Destination data shape.")
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:55: TracerWarning: Converting a tensor to a Python list might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  out_size = tuple(params["output_size"][0].tolist())
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:59: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  x1 = int(params["src"][i, 0, 0])
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:60: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  x2 = int(params["src"][i, 1, 0]) + 1
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:61: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  y1 = int(params["src"][i, 0, 1])
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:62: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  y2 = int(params["src"][i, 3, 1]) + 1
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/utils/image.py:222: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if input.numel() == 0:
/home/duc/anaconda3/lib/python3.9/site-packages/kornia/geometry/transform/affwarp.py:574: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if size == input_size:
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
n/_2d/geometric/resize.py:64:0
  %onnx::Slice_17 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={1}, onnx_name="Constant_16"]() # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:64:0
  %input : Float(1, 3, 270, 480, strides=[3, 1, 1440, 3], requires_grad=0, device=cpu) = onnx::Slice[onnx_name="Slice_17"](%onnx::Slice_13, %onnx::Slice_15, %onnx::Slice_16, %onnx::Slice_14, %onnx::Slice_17) # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:64:0
  %onnx::Slice_20 : Long(4, strides=[1], device=cpu) = onnx::Shape[onnx_name="Shape_18"](%input) # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Slice_21 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={0}, onnx_name="Constant_19"]() # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Slice_22 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={0}, onnx_name="Constant_20"]() # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Slice_23 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={2}, onnx_name="Constant_21"]() # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Concat_24 : Long(2, strides=[1], device=cpu) = onnx::Slice[onnx_name="Slice_22"](%onnx::Slice_20, %onnx::Slice_22, %onnx::Slice_23, %onnx::Slice_21) # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Resize_26 : Long(4, strides=[1], device=cpu) = onnx::Concat[axis=0, onnx_name="Concat_23"](%onnx::Concat_24, %onnx::Concat_47) # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Resize_27 : Tensor? = prim::Constant() # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Resize_28 : Tensor? = prim::Constant() # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Reshape_29 : Float(*, *, *, *, strides=[36720, 12240, 170, 1], requires_grad=0, device=cpu) = onnx::Resize[coordinate_transformation_mode="align_corners", cubic_coeff_a=-0.75, mode="linear", nearest_mode="floor", onnx_name="Resize_24"](%input, %onnx::Resize_27, %onnx::Resize_28, %onnx::Resize_26) # /home/duc/anaconda3/lib/python3.9/site-packages/torch/nn/functional.py:3938:0
  %onnx::Reshape_31 : Long(3, strides=[1], device=cpu) = onnx::Constant[value=   3   72  170 [ CPULongType{3} ], onnx_name="Constant_25"]() # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:63:0
  %onnx::Expand_32 : Float(*, *, *, strides=[12240, 170, 1], requires_grad=0, device=cpu) = onnx::Reshape[onnx_name="Reshape_26"](%onnx::Reshape_29, %onnx::Reshape_31) # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:63:0
  %onnx::Reshape_34 : Float(*, *, *, strides=[12240, 170, 1], requires_grad=0, device=cpu) = onnx::Expand[onnx_name="Expand_27"](%onnx::Expand_32, %onnx::Expand_48) # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:63:0
  %onnx::ScatterND_45 : Float(*, *, *, *, device=cpu) = onnx::Reshape[onnx_name="Reshape_28"](%onnx::Reshape_34, %onnx::Reshape_53) # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:63:0
  %46 : Float(*, *, *, *, strides=[36720, 12240, 170, 1], requires_grad=0, device=cpu) = onnx::ScatterND[onnx_name="ScatterND_29"](%onnx::ScatterND_3, %onnx::ScatterND_50, %onnx::ScatterND_45) # /home/duc/anaconda3/lib/python3.9/site-packages/kornia/augmentation/_2d/geometric/resize.py:63:0
  return (%46)

### Environment

```shell
- PyTorch Version 3.9
- OS  Linux
```

**Comment by duc12111** [comment]
Hi @edgarriba , Do you think this can be fixed? This shape inference would be required by onnx for some layers like adaptive_avg_pool2d.

**Comment by edgarriba** [comment]
we should put a test with onnx and debug /cc @shijianjian  @ducha-aiki

**Comment by duc12111** [comment]
Hi @edgarriba, is anyone working on this? Otherwise, I can work on it a little bit. Any suggestions on the test onnx?

**Comment by shijianjian** [comment]
> Hi @edgarriba, is anyone working on this? Otherwise, I can work on it a little bit. Any suggestions on the test onnx?

Please go for it. One thing regarding ONNX is that we need to decide if we want it to be dynamic. If we export the ONNX model for 32x3x128x128, I am thinking if we shall allow the dynamic shape here.

**Comment by duc12111** [comment]
@shijianjian  Do you mean we should have the option to have the shape dynamic? Since in my use case, I would like to have the output—shape well-defined since this information could be used in the next layer.

**Comment by shijianjian** [comment]
> @shijianjian Do you mean we should have the option to have the shape dynamic? Since in my use case, I would like to have the output—shape well-defined since this information could be used in the next layer.

If it is not too complex, the dynamic shape would be preferred. But with a fixed shape can be a start point.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
